File: server/app/schemas/standing.py
from typing import Literal
from pydantic import BaseModel

class Standing(BaseModel):
    team_id: int
    team_city: str
    team_name: str
    conference: Literal["East", "West"]
    division: str
    conference_rank: int
    wins: int
    losses: int
    win_pct: float
    streak: str
    last_ten: str

File: client/src/types/Standing.ts
import type { Team } from "./Team";
import type { Conference } from "./Conference";

export interface Standing {
    team: Team;
    conference: Conference;
    division: string;
    conference_rank: number;
    wins: number;
    losses: number;
    win_pct: number;
    streak: string;
    last_ten: string;
}